Nouvelle interface pour la liste des tuto

artragis opened the pull request:
| Q                                   | R
| ----------------------------------- | -------------------------------------------
| Type de modification                | évolution
| Ticket(s) (_issue(s)_) concerné(s)  | No ticket

Note: le nouveau design est un peu "caché" pour le moment, mais est disponible dans http://0.0.0.0:8000/publications/ . 

# Reste à faire

## Général

+ [x] Flux RSS s'adaptant à la catégorie
+ [x] Urls pour inclure catégories et sous-catégories (ce qui sera probablement plus simple pour gérer les templates)
+ [x] Liste des contenus correspondants à ce qui est demandé (pour le moment, ça ne fonctionne pas, on a juste la liste des *x* derniers contenus dans chaque colonne)
+ [x] Recherche
+ [x] Variables en dur (des variables sont en fait déjà définies dans le `settings.py` mais pas dans le code)
+ [x] Gestion des images
+ [x] Rajouter des petites infobules sur les liens, pour les catégories aux titres trop long
+ [x] Sur quoi pointe les liens "plus de tutoriels" et "plus d'articles" ?
+ [x] Gestion des listes vides: est ce que ça a du sens d'avoir le bouton "plus de tutoriels" quand la liste est vide ?
+ [x] Corriger le *dropdown*

## Page de premier niveau

+ [x] "Plus de tutoriels" et "plus d'articles" ne fonctionnent pas
+ [x] Administration pour la gestion des mises en avant 
+ [x] Bonne url pour le forum bêta

## Page de second niveau

+ [x] "Plus de tutoriels" et "plus d'articles" ne fonctionnent pas.

## Page de troisième niveau

+ [x] Faire la page de troisième niveau

## Page de 4e niveau

+ [x] Faire la page de 4e niveau

# Relecture de code

Elle est primordiale pour plusieurs raisons : 

- entre le début du développement et aujourd'hui beaucoup de choses ont changé et le dernier rebase était plein de problèmes côté webdesign
- il ne faut pas qu'on crée des goulets d'étranglement

# QA

Faut visiter la page de contenu et voir ce qu'on a mal fait et/ou oublié

artragis committed:
début du travail : on split les templates

artragis committed:
début du travail : on split les templates

artragis committed:
top tuto

artragis committed:
top tuto

artragis committed:
some fixes

artragis committed:
some fixes

abdelaz3r committed:
nouveau composant css : linkbox

abdelaz3r committed:
nouveau composant css : linkbox

abdelaz3r committed:
création grossière du template

abdelaz3r committed:
création grossière du template

abdelaz3r committed:
mise en place de la structure du niveau 0

abdelaz3r committed:
mise en place de la structure du niveau 0

artragis committed:
some adapation of mainpage

artragis committed:
some adapation of mainpage

abdelaz3r committed:
modification CSS

abdelaz3r committed:
modification CSS

abdelaz3r committed:
correction HTML page niveau 0

abdelaz3r committed:
correction HTML page niveau 0

artragis committed:
little fix

artragis committed:
little fix

abdelaz3r committed:
élément statique : mise à jour des liens et des descriptifs

abdelaz3r committed:
élément statique : mise à jour des liens et des descriptifs

abdelaz3r committed:
premier travail sur le header

abdelaz3r committed:
premier travail sur le header

abdelaz3r committed:
travail sur le header + adaptation de la page pour les contenus

abdelaz3r committed:
travail sur le header + adaptation de la page pour les contenus

artragis committed:
ajoute des données dans la vue

artragis committed:
ajoute des données dans la vue

abdelaz3r committed:
les flux sont désormais à droite du titre

artragis committed:
some fixes

artragis committed:
new list

artragis committed:
typo

abdelaz3r committed:
préparation du deuxième niveau hiérarchique

artragis committed:
lien vers les forums

artragis committed:
comptage des contenus.

abdelaz3r committed:
ajout du composant scss more-link

abdelaz3r committed:
ajout des liens plus de tuto/article sur le premier niveau

abdelaz3r committed:
ajout des liens plus de tuto/article sur le deuxième niveau

artragis committed:
fix pep8 and doubly

artragis committed:
unimport

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+
+                <a href="{% url 'content:helps' %}?type=tuto" class="body">
+                    <p>Les auteurs ont souvent besoin d'un peu d'aide pour amener leur contenu jusqu'à la publication.</p>
+                    <p>Dessiner une icône ou des illustrations, corriger un texte ou encore devenir co-auteur vous interesse ?</p>
```
"intéresse" (avec un accent)

abdelaz3r replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+        <div class="content-linkbox-list">
+            <div class="linkbox-item secondary">
+                <a href="{{ beta_forum.get_absolute_url }}" class="head">
+                    <h3>Donner son avis</h3>
```
Plutôt "votre avis" par cohérence avec le deuxième bloc ?

abdelaz3r replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+                </a>
+
+                <a href="{{ beta_forum.get_absolute_url }}" class="body">
+                    <p>Les contenus presque terminés nécessitent souvent une relecture afin de peaufiner le fond comme la forme.</p>
```
"afin d'*en* peaufiner" ?

artragis replied:
je sais pas

AmauryCarrade replied:
Je trouve aussi qu'« afin d'en peaufiner » sonne mieux dans cette phrase.

abdelaz3r replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+
+                <a href="{{ beta_forum.get_absolute_url }}" class="body">
+                    <p>Les contenus presque terminés nécessitent souvent une relecture afin de peaufiner le fond comme la forme.</p>
+                    <p class="right">Consulter les contenus en bêta.</p>
```
J'aurais tendance à enlever le point ici.

abdelaz3r replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+                </div>
+
+                <div class="body">
+                    <p>Vous avez un savoir à partager, vous vous sentez pédagogue, pourquoi ne pas commencer l'écriture d'un tutoriel ?</p>
```
Pour rester générique, je me contenterais de "contenu" ici.

abdelaz3r replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+                <a href="{% url 'content:helps' %}?type=tuto" class="body">
+                    <p>Les auteurs ont souvent besoin d'un peu d'aide pour amener leur contenu jusqu'à la publication.</p>
+                    <p>Dessiner une icône ou des illustrations, corriger un texte ou encore devenir co-auteur vous interesse ?</p>
+                    <p class="right">Parcourir les demandes d'aides.</p>
```
Idem ici (pas de point)

abdelaz3r replied:
done

gllmc commented on templates/tutorialv2/list_page_elements/list_of_content.html:
```diff
@@ -0,0 +1,40 @@
+{% endif %}
+    {% if public_contents %}
+        <div class="content-item-list">
+            {% if current_content_type == "ARTICLE" %}
```
#4328 ayant été mergé, attention ici à bien mettre le `or current_content_type == "OPINION"`.

artragis replied:
toutafé :)

artragis replied:
done

gllmc commented on templates/tutorialv2/list_page_elements/side_bar.html:
```diff
@@ -0,0 +1,119 @@
+            {% url "tutorial:list" %}
+        {% elif current_content_type == "OPINION" %}
+            {% url "opinion:list" %}
+        {% elif current_content_type == None %}
```
`elif not current_content_type` me paraît mieux à titre personnel, mais c'est subjectif.

artragis replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+    - liste des 6 derniers tutos en une colonne
+    - liste des 6 tutos les plus commentés en une colonne
+    - bloc de envie de s'investir
+{% endcomment %}
```
Il faudra ensuite enlever ces commentaires en français (ou les traduire).

artragis replied:
Done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+            </div>
+            <div class="title">
+                <h1>Contenus</h1>
+                {{ content_count }} contenus
```
`content_count` n'existe pas chez moi et i18n sur les lignes 17 et 18.

artragis replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+                <label for="id_q" class="control-label ">Recherche</label>
+                <input id="id_q" maxlength="150" name="q" required="required" type="search">
+                <input type="hidden" name="models" value="tutorial,article">
+                <button class="btn ico-after ico-search" type="submit" title="Rechercher"></button>
```
i18n (lignes 22 et 25)

artragis replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+
+<div class="flexpage-wrapper">
+    <section>
+        <h2 itemprop="name">À la une</h2>
```
i18n

artragis replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+    </section>
+
+    <section>
+        <h2 class="inline" itemprop="name">Domaines de savoir</h2>
```
i18n

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+                <a href="{% url "article:feed-atom" %}" class="btn btn-grey">Atom</a>
+            </h2>
+            {% include 'tutorialv2/list_page_elements/list_of_content.html' with public_contents=selected_contents col_number=1 %}
+            <a href="#" class="more-link">Plus d'articles</a>
```
i18n (lignes 51, 56, 61 et 66)

artragis replied:
done

gllmc commented on templates/tutorialv2/view/all_content_page.html:
```diff
@@ -0,0 +1,119 @@
+                <a href="{% url 'content:helps' %}?type=tuto" class="body">
+                    <p>Les auteurs ont souvent besoin d'un peu d'aide pour amener leur contenu jusqu'à la publication.</p>
+                    <p>Dessiner une icône ou des illustrations, corriger un texte ou encore devenir co-auteur vous interesse ?</p>
+                    <p class="right">Parcourir les demandes d'aides.</p>
```
Il manque l'i18n dans le bloc "Envie de s'investir ?".

artragis replied:
done

gllmc commented on zds/tutorialv2/managers.py:
```diff
@@ -136,7 +177,7 @@ def transfer_paternity(self, unregistered_user, replacement_author, gallery_clas
                     versioned = content.load_version()
                     title = versioned.title
-                    introduction = _(u'[[i]]\n|Ce contenu a été rédigé par {} qui a quitté le site.\n\n')\
+                    introduction = '[[i]]\n|Ce contenu a été rédigé par {} qui a quitté le site.\n\n'\
```
Pourquoi cette modification ?

artragis replied:
done

gllmc reviewed (commented):
:+1: 

Je trouve que le rendu est très bon et que ça va bien permettre d'améliorer la présentation des contenus qu'on a.

Voici les quelques problèmes que j'ai relevés.

artragis commented:
https://www.pixenli.com/images/1493/1493838040036010300.png

artragis commented:
Pour l'image, j'ai trouvé, ligne 6 de "all_content_page" : `<img src="http://127.0.0.1:8000/static/images/home-clem.png">`. @abdelaz3r il faudra que tu expliques ce que tu voulais mettre.

artragis committed:
review

Conflicts:
	templates/tutorialv2/list_page_elements/side_bar.html
	templates/tutorialv2/view/all_content_page.html

AmarOk1412 commented:
Pour l'instant j'ai une erreur sur les lignes ` <p>{% trans 'Vous avez un savoir à partager, vous vous sentez pédagogue, pourquoi ne pas commencer l'écriture d'un tutoriel ?' %}</p>` et `<h2 class="inline">{% trans 'Envie de s'investir ?' %}</h2>` : `TemplateSyntaxError: Only options for 'trans' are 'noop', 'context "xxx"', and 'as VAR'.`

Sinon, j'ai des carrés blancs dans les blocs en bas (les span icon) et les boutons "Plus de tutoriels/articles" sont peut etre trop différent de ce qu'on trouve dans le reste du site.

Les pages "contenus/?category=arts-graphisme-et-multimedia" sont vides (normal, reste à faire)

Changer l'icone de clem en haut  ?

Changer la top bar.

Mais ça a l'air très bien parti :)

AmarOk1412 commented on assets/scss/components/_linkbox.scss:
```diff
@@ -0,0 +1,173 @@
+            position: relative;
+            color: white;
+            text-decoration: none;
+&:after {
```
Juste bien aligner ici pour que ça fasse plus propre (et l'accolade de fin), sinon le reste du code m'a l'air correct !

artragis committed:
fix template error

artragis commented:
@AmarOk1412 j'ai fixé le pb du template.

>Changer la top bar.

c'est à dire?

>Changer l'icone de clem en haut ?

j'attends le retour @abdelaz3r  pour ça.

>Les pages "contenus/?category=arts-graphisme-et-multimedia" sont vides (normal, reste à faire)

Heu, vide, c'est à dire? pas de tuto/article ou pas de design? @abdelaz3r tu peux jeter un coup d'oeil?

artragis committed:
escape

AmarOk1412 commented:
> @AmarOk1412 j'ai fixé le pb du template.

yeap c'est bon

Pour la top bar je parlais de ce point : https://zestedesavoir.com/forums/sujet/4856/discussions-sur-le-design-de-zestedesavoir/?page=3#p141039 *Sur la topbar, il n’y a plus qu’un bouton contenu en lieu et place des boutons tutos et articles;*

Pour http://127.0.0.1:8000/contenus/?category=communication-et-management, ce que j'ai c'est  : (avec 1 contenu logiquement)
![2017-05-06-16 49 03-screenshot](https://cloud.githubusercontent.com/assets/1888628/25775809/f691af36-327b-11e7-8ddd-32d43d47eaba.png)

Le petit log : 
> [06/May/2017 22:48:59] "GET /contenus/?category=communication-et-management HTTP/1.1" 200 30500
> Not Found: /favicon.ico
> [06/May/2017 22:49:47] "GET /favicon.ico HTTP/1.1" 404 16706
> - Broken pipe from ('127.0.0.1', 43554)

artragis commented:
>Sur la topbar, il n’y a plus qu’un bouton contenu en lieu et place des boutons tutos et articles;

C'est voulu.

>ce que j'ai c'est : (avec 1 contenu logiquement)

étrange.

abdelaz3r commented:
Yep. Déso beaucoup de trucs ces temps. Je vais jeter un oeil à tout ça !

pierre-24 commented:
20 ans plus tard, j'apparais: à priori, je vois pas encore la partie sur la recherche implémentée, du coup ou est ce que vous avez besoin d'un coup de patte à ce niveau là ?

abdelaz3r commented:
Oui très clairement. En fait il faudrait que le formulaire de recherche renvoi un résultat spécifique selon ou on est.

Sur la home des contenus (all_content_page.html) il faut chercher dans tout les contenus (articles + tutos), sur les pages de catégorie (one_theme_content_page.html) il faut chercher dans les contenus (articles + tutos) appartenant à la catégorie (variable "theme" dans le template).

abdelaz3r committed:
correction des carrés blancs sur certains liens

abdelaz3r committed:
correction de structures dans la page d'une catégorie

pierre-24 commented:
Normalement, c'est facile, faut juste que je réfléchisse un tout petit peu comment forcer le type et la catégorie, du coup :)

abdelaz3r commented:
# Truc encore à faire

Voilà un petit retour sur ce qu'il reste à faire. Comme dit à @artragis je ne m'occupe que des templates et du front. Donc il y a des petits trucs intéressants à faire pour me débloquer l'avancement.

## Général

- [x] Faire un nouveau "ancien" template pour les tribunes qui ne partagent plus ce template.

## all_content_page.html

- [x] Charger les contenus "à la une" (et faire qqch dans l'admin pour pouvoir choisir lesquels seront à la une).
- [x] Dans le bloc donner son avis : mettre l'url vers le forum de béta.

## one_theme_content_page.html

- [x] Effectuer la recherche uniquement sur la catégorie (@pierre-24).
- [x] N'afficher que 10 derniers articles.
- [x] N'afficher que 10 derniers tutoriels.
- [x] N'afficher que les articles appartenant à "theme".
- [x] N'afficher que les tutoriels appartenant à "theme".
- [x] Modifier le link du flux rss/atom article pour filtrer selon la catégorie.
- [x] Modifier le link du flux rss/atom tutoriel pour filtrer selon la catégorie.

## Dernier niveau hiérarchique (new_template.html)

De manière général le dernier niveau sera assez simple visuellement. Il y aura toujours le bloc bleu en haut (mais sans la recherche et qui du coup ne sera qu'un fil d'ariane avec des options filtre/tri). Je vais faire une maquette pour que ça marche. Donc il faudrait créer cette page avec une url, template et tout et tout. Je m'occupe de la remplir.

Après en fait cette page elle va *juste* afficher une liste **PAGINEE** de contenu avec "pas mal" d'option de filtre et de tri. Il faut donc modifier list_of_content.html pour permettre plus d'option. Notamment une *limite de contenu à charger*. De plus il faudrait une variable de filre et de tri.

Donc on pourrait spécifier dans les filtres :

- le type (article/tuto)
- la catégorie (une des 4)
- la sous-catégorie

Et dans le tri :

- par date (publication asc/desc)
- alphabétique (à voir)
- par nombre de vue (pour plus tard)

# Remarques générales pour les URL

De manière général il faut réfléchir à l'arborescence. Je trouve pas le mieux de mettre l'URl de la page d'une catégorie comme ça `http://127.0.0.1:8000/contenus/?theme=Sciences%20humaines%20et%20sociales`, je verrai plutôt comme ça (mais c'est à discuter) : `http://127.0.0.1:8000/contenus/sciences-humaines-et-sociales` (avec un slug et pas un truc avec des majuscules et des espaces).

De plus il faudrait penser au nom et à l'affichage du dernier niveau hiérarchique. Qui peut afficher un peu ce qu'il veut.

abdelaz3r committed:
correction du texte des boutons 'plus de contenu'

abdelaz3r committed:
correction de typos

pierre-24 commented:
Et du coup, les résultats de la recherche devraient-il s'afficher dans les templates que tu as prévues (un peu plus chaud) ou comme les résultats de recherches comme ils sont proposés maintenant ?

abdelaz3r commented:
Comme les résultats actuels

qwertygc commented:
Possible de voir un screenshot pour voir le résultat actuel ?

abdelaz3r commented:
La page de premier niveau:

![p1-1](https://cloud.githubusercontent.com/assets/3975144/26449766/64b6d32e-4154-11e7-8d69-a7e5725a5874.png)
![p1-2](https://cloud.githubusercontent.com/assets/3975144/26449780/743cf706-4154-11e7-8480-b61a473b33df.png)

La page de deuxième niveau:

![p2-1](https://cloud.githubusercontent.com/assets/3975144/26449786/7c047e1e-4154-11e7-99b0-5da82f3d471c.png)

qwertygc commented:
Et si on rajoutait une troisième colonne pour les billets ?

abdelaz3r commented:
Absolument pas ! Les tribunes et le restes des contenus sont pas du tout la même chose.

La, l'idée étant de rassembler en une seule page tout les contenus validés et de pouvoir parcourir comme dans une bibliothèque des contenus de zeste. Les tribunes c'est un tout autre truc.

qwertygc commented:
Donc si je comprends bien, ça permettra d'introduire les parcours ?

abdelaz3r commented:
Bah a plus long terme oui, mais ce n'est pas vraiment le sujet ici. On peut en discuter sur le forum, de voir comment la présentation des contenus peut évoluer. Ici c'est une PR qui propose une interface simple, unifiée des contenus. Il y a bien sûr des évolutions possibles mais je ne pense pas qu'il faille en discuter ici.

abdelaz3r commented:
@artragis : tu as du temps pour avancer sur ces points ces prochains jours ?

abdelaz3r committed:
squelette du template de dernier niveau

abdelaz3r committed:
Create _linkbox.scss

artragis commented:
>N'afficher que 10 derniers articles.

je check ça rapidement, mais c'est pas hyper compliqué. 

>Faire un nouveau "ancien" template pour les tribunes qui ne partagent plus ce template.

ça je te laisse le faire, mais un petit copier/coller du template qui est sur le repo actuel devrait suffir, non?

>N'afficher que les articles appartenant à "theme".

le back est prêt, à toi de gérer.

>Modifier le link du flux rss/atom article pour filtrer selon la catégorie.

Je vais vérifier, mais il me semble que c'est aussi de ton côté.

artragis committed:
fix numbers

artragis commented:
J'ai fait quelques modifs sans avoir eu le temps de les tester. Mais en attendant, @abdelaz3r le nombre de tuto/article devrait être bon

artragis committed:
typo

artragis committed:
typo

pierre-24 commented:
*édité*

pierre-24 committed:
Add (sub)category to search

pierre-24 committed:
a whole bunch of new stuffs

pierre-24 commented:
Bon, du coup, ce que j'ai fait: 

+ J'ai gardé l'ancienne page: à chaque fois qu'on clique sur "tout les ...", on retombe sur cette liste à l'ancienne. Comme ça, ça fonctionne toujours pour les billets, et on est tranquille.
+ J'ai implémenté la plupart des trucs, entre autre une url pour thème/catégorie. Du coup, j'ai été obligé de créer un nouveau point d'entrée, `/publications/`. Parce que `/contenus/` contient en fait déjà beaucoup de trucs en rapport avec la gestion des contenus, et qu'on a pas été très intelligents quand on a fait ça.
+ J'ai pas encore chipoté au dropdown.
+ J'ai renommé et factorisé. Désolé pour les noms originaux ^^

pierre-24 committed:
debug: now fields contain slugs

pierre-24 commented on zds/settings.py:
```diff
@@ -509,7 +509,11 @@
+        'sec_per_minute': 1500,
+        'list': 'new',  # put 'old' if you want to go back to old way of displaying tutorial
+        'selected_content_method_name': 'get_most_commented_list',  # allow to customize "top tutorial" appears
+        'content_per_theme': 10
```
*Note to self*: ces paramètres ne sont utilisés nul part. Il faudrait ^^ (mais qu'est ce que `'selected_content_method_name'` et `'list'` sont sencé faire ?!?)

pierre-24 commented:
Question: 

Comment adapter le design pour les titres un peu trop long ?

![screenshot from 2017-08-03 22 31 27](https://user-images.githubusercontent.com/7889675/28942436-9fe34ac6-789b-11e7-9e51-0bb6b6a45e58.png)

pierre-24 committed:
we need beta forum as well

motet-a commented:
@pierre-24 Avec quelque chose comme ça ?

![image](https://user-images.githubusercontent.com/15378830/28972546-5d806cd2-7930-11e7-945b-592da90cd52b.png)

Je pense qu’il faudrait aussi mettre un petit effet léger au survol des liens de ce fil d’ariane.

motet-a commented:
Petit problème de marge sur mobile :

![image](https://user-images.githubusercontent.com/15378830/28972806-61e85d7e-7931-11e7-8bd4-584f7093c586.png)

vhf committed:
WIP: suite et fin

motet-a commented:
Corrige le petit problème de marge que j’ai signalé plus haut :

```diff
diff --git a/assets/scss/pages/_flexpage.scss b/assets/scss/pages/_flexpage.scss
index c5872341..dce5fb60 100644
--- a/assets/scss/pages/_flexpage.scss
+++ b/assets/scss/pages/_flexpage.scss
@@ -198,7 +198,7 @@ $content-width: 1145px;
 
         .flexpage-column {
             display: block;
-            margin-left: 10px;
+            margin-left: 0;
 
             section {
                 width: 100%;
```

vhf committed:
Corrige les liens vers les vues articles/tutos

pierre-24 commented on zds/tutorialv2/views/views_published.py:
```diff
@@ -404,131 +404,95 @@ class ListOpinions(ListOnlineContents):
+                self.request.GET.get('type', False) or \
+                self.request.GET.get('tags', False):
+            self.level = 4
+            raise NotImplemented("filter view, cf. TODO.md at repository root")
```
NotImplementedError() en fait (je me fais avoir à chaque fois)

vhf replied:
yep !

pierre-24 commented on zds/tutorialv2/views/views_published.py:
```diff
@@ -404,131 +404,95 @@ class ListOpinions(ListOnlineContents):
+            # get categories and subcategories
+            context['categories'] = list(Category.objects.order_by('position').all())
+            for category in context['categories']:
+                category.subcategories = list(
```
Peut être remplacé par `category.subcategories = category.get_subcategories()` (mais ça demande de chipoter la template un peu).

vhf replied:
Je préfère éviter pour le moment

pierre-24 commented on zds/tutorialv2/views/views_published.py:
```diff
@@ -404,131 +404,95 @@ class ListOpinions(ListOnlineContents):
+                pks = [c['subcategory__pk'] for c in category.subcategories]
+                category.contents_count = PublishedContent.objects\
+                    .published_contents()\
+                    .filter(content__subcategory__in=pks)\
```
`content_subcategory__in=category.subcategories`, du coup

vhf replied:
Merci !

pierre-24 commented on zds/tutorialv2/views/views_published.py:
```diff
@@ -404,131 +404,95 @@ class ListOpinions(ListOnlineContents):
         context['last_tutorials'] = PublishedContent.objects.get_recent_list(
-            content_type='TUTORIAL',
-            subcategories=[self.object])[:5]
+            **dict(content_type='TUTORIAL', **recent_kwargs)
```
Joli <3

vhf commented:
@abdelaz3r 
* j'ai rebasé donc faudra bien tirer avant de bosser sur le CSS.
* tu peux me faire une maquette pour le lvl `4` ?

pierre-24 commented on zds/tutorialv2/urls/urls_articles.py:
```diff
@@ -26,6 +28,6 @@
 
     # Listing
-    url(r'^$', ListArticles.as_view(), name='list'),
+    url(r'^$', RedirectView.as_view(url=reverse('publication:list'))),
```
`, permanent=True`

abdelaz3r commented:
@vhf : tu as besoin d'une maquette ? Tu veux juste pas créer le template et je le met en forme ?

pierre-24 commented:
*edit*: appliqué :)

vhf committed:
WIP

abdelaz3r committed:
ajout de l'icone de la bibliothèque

vhf committed:
Level 4: WIP

abdelaz3r committed:
ajout de style pour le 4ème niveau (browse)

vhf committed:
Level 4: Implémentation des filtres

vhf committed:
Level 4: Corrections

abdelaz3r committed:
correction visuelle en-tête

vhf committed:
Level 4: Corrections

vhf committed:
Affiche 'Derniers x' seulement si x il y a

pierre-24 commented:
C'est génial <3

Mais ... 

![](https://ihatehongkong2.files.wordpress.com/2014/06/4488876837_8d3da2423a_z.jpg)

Rapport de QA:

- [x] Dans le dropdown, chez moi, la flèche est très proche du mot "bibliothèque":

![screenshot from 2017-08-06 13 24 52](https://user-images.githubusercontent.com/7889675/29002801-022e3f22-7aab-11e7-91d4-174d9bd9454e.png)

- [x] Sur le screenshot ci-dessus, on vois que "bibliothèque" ne se sélectionne pas lorsqu'on est dans la bibliothèque (voir ci-dessous pour les tribunes):

![screenshot from 2017-08-06 13 44 21](https://user-images.githubusercontent.com/7889675/29002912-7638ce80-7aad-11e7-8f9d-32704d6bdb79.png)


- [x] Les liens dans le dropdown ont toujours l'ancien format en "filtre" (d'où 404)
- [ ] Quand on parcourt une sous-catégorie vide, il manque un "aucun contenu" (j'ai pas testé sur une catégorie vide):

![screenshot from 2017-08-06 13 30 48](https://user-images.githubusercontent.com/7889675/29002822-7b92c61c-7aab-11e7-9afe-74c3c8c2ba3f.png)

- [x] Ça serait éventuellement bien que sur la page "niveau 4", domaine et catégorie soient clicables.
- [x] Je sais pas si c'est vous (j'ai pas ça sur `dev`, en tout cas), mais y'a plus "zeste de savoir" dans l'atribut titre, et ce sur toutes les pages:

![screenshot from 2017-08-06 13 35 14](https://user-images.githubusercontent.com/7889675/29002854-2bf25324-7aac-11e7-81cf-58d4a508d44e.png)

- [x] Les flux RSS des catégories (`?category=xxx`) sont vides (mais ceux généraux et des sous-catégories fonctionnent)

motet-a commented:
![image](https://user-images.githubusercontent.com/15378830/29002929-b66ffea6-7aad-11e7-9a50-e2d1c1a7a676.png)

Hey les gens, tenez pas compte de ce problème de flèche et ne touchez à rien, j’ai un fix : #4440

abdelaz3r committed:
ajout des infobulles sur les titres des catégories

abdelaz3r committed:
les options du niveau 4 sont cliquables

abdelaz3r commented:
> Ça serait éventuellement bien que sur la page "niveau 4", domaine et catégorie soient clicables.

Done !

vhf committed:
Corrige les liens du menu

vhf committed:
Corrige l'onglet du menu 'current' state

motet-a commented:
![image](https://user-images.githubusercontent.com/15378830/29003330-1b9b22fe-7ab5-11e7-9a78-d47b80235260.png)

Sur mobile, ça serait bien de mettre moins de marge et une ellipse : « Biblio… »

motet-a commented:
![image](https://user-images.githubusercontent.com/15378830/29003353-6a26abfa-7ab5-11e7-8c25-f4a1108ec5e9.png)

Voilà ce que ça donne avec un vrai navigateur.

motet-a commented:
À part ça j’ai bien essayé sur IE nottament et je n’ai rien à rajouter à la review de @pierre-2.

vhf committed:
Corrige la génération de la topbar

vhf committed:
Corrige le flux des contenus d'une catégorie

vhf committed:
Corrige liens items->cat; corrige no content

vhf committed:
Corrige l'index Bibliotheque
